### PR TITLE
Add support for freezing time to facilitate testing with time

### PR DESCRIPTION
**Pull request by mumia**
Based on code found in https://github.com/ssoroka/ttime

### CODE REVIEW

File: carbon.go [hunk 1]
```diff
@@ -83,6 +89,62 @@ func NewCarbon(t time.Time) *Carbon {
...
+
+// After will be behave like time.After unless time has been frozen
+// If time is frozen it will add the expected delay and immediately send the frozen time on the returned channel
+func After(d time.Duration) <-chan time.Time {
+	if isTimeFrozen {
+		currentFrozenTime = currentFrozenTime.Add(d)
+		c := make(chan time.Time, 1)
+		c <- currentFrozenTime
+		return c
+	} else {
```
Comment (mlimaloureiro) [inline carbon.go]: remove the else and just return

File: carbon.go [hunk 2]
```diff
@@ -83,6 +89,62 @@ func NewCarbon(t time.Time) *Carbon {
...
+	if isTimeFrozen {
+		c := make(chan time.Time, 1)
+		go func() {
+			for {
+				currentFrozenTime = currentFrozenTime.Add(d)
+				c <- currentFrozenTime
+			}
+		}()
+		return c
+	} else {
```
Comment (mlimaloureiro) [inline carbon.go]: same here

File: carbon.go [hunk 3]
```diff
@@ -83,6 +89,62 @@ func NewCarbon(t time.Time) *Carbon {
...
+		return time.Tick(d)
+	}
+}
+
+// Sleep will be behave like time.Sleep unless time has been frozen
+// If time is frozen it will add the expected sleep delay and return immediately
+func Sleep(d time.Duration) {
+	if isTimeFrozen && d > 0 {
+		currentFrozenTime = currentFrozenTime.Add(d)
+	} else {
```
Comment (mlimaloureiro) [inline carbon.go]: same here

File: carbonfreeze_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
+package carbon_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
```
Comment (jsaguiar) [inline carbonfreeze_test.go]: remove nl

**Comment by mumia** [comment]
Closed in favor of #34

**Pull request closed without merging**